models/articles.model.js

```diff
@@ -35,5 +35,38 @@ async function selectArticles(sort_by='created_at') {
     }
 }
 
+async function selectCommentsByArticleId(id) {
+    try {
+        const comments = await db.query(`
+        SELECT * FROM comments WHERE article_id=$1 ORDER BY created_at DESC;`, [id]);
+        return comments.rows;
+    } catch (error) {
+        throw error
+    }
+}
+
+async function checkArticleIdExists(id) {
+    try {
+        const article = await db.query(`
+        SELECT * FROM articles WHERE article_id=$1`, [id]);
+        if (article.rows.length === 0) {
+            return Promise.reject({status: 404, message: 'article not found'})
+        }
+    } catch (error) {
+        throw error
+    }
+}
+
+async function insertCommentByArticleId(username, body, id) {
+    try {
+        const comment = await db.query(`
+        INSERT INTO comments (body, author, article_id) VALUES ($1, $2, $3) RETURNING *;
+        `, [body, username, id]);
+        return comment.rows[0];
```

Comment:
if rows[0] here was empty we could promise reject as it wouldnt exist. This means you wouldnt have to check it exists in the promise.all :) up to you !